src/strategies/scraper.ts

```diff
@@ -0,0 +1,99 @@
+import { ElementHandle } from 'puppeteer';
+import { sleep } from '../util/sleep';
+import { LikeOptions } from '../luckybot';
+import { PageProvider } from '../page'
+import { HOST } from '../config';
+import { Session } from '../session';
+
+export class Scraper {
+  isLoggedIn: boolean = false;
+  pageProvider: PageProvider;
+  session: Session;
+
+  constructor() {
+    this.pageProvider = new PageProvider();
+    this.session = new Session();
+  }
+
+  async login(userName: string, password: string) {
+    const cookies = await this.session.login(userName, password);
+    const page = await this.pageProvider.getPage();
+
+    await page.setCookie(...cookies)
+  }
+
+  private async getPhotos(hashtag: string) {
+    const page = await this.pageProvider.getPage();
+
+    await page.goto(`${HOST}/explore/tags/${hashtag}/`);
+    const article = await page.waitForSelector('article');
+    const photos = await article.$$('a');
+    
+    return photos;
+  }
+
+  private async closePhotoDetail() {
+    const page = await this.pageProvider.getPage();
+    await sleep(1000);
+    // TODO: make generic + move somewhere else
+    const closeButtonClassName = await page.evaluate(() => {
+      const closeButton = Array.from(document.querySelectorAll('button')).find(button => button.textContent === 'Close');
+      return closeButton && closeButton.className;
+    });
+
+    if (closeButtonClassName) {
+      const closeButton = await page.$(`.${closeButtonClassName}`);
+      closeButton && closeButton.click();
+    }
+
+    await sleep(1000);
+  }
+
+  // TODO: return boolean liked?
+  private async likePhoto(photo: ElementHandle<Element>, delay: number): Promise<void> {
+    const page = await this.pageProvider.getPage();
+
+    await sleep(delay);
```

Comment:
Are these delays just to look more real or to wait for stuff with random interval?

Reply:
To make it look real, otherwise it's too fast